### DISCUSSION TITLE
Scene2D builder asks for explicit reciever

### DISCUSSION THREAD

**Discussion by Divelix** [libktx/ktx#301] [Q&A] [answered]
I keep fixing things after update to `b6` ))). Now AS shows error in this place, that worked well before update:
```kotlin
class ModIcon(val mod: Mod, ...) {
    val lvlDrawable: Drawable = ...
    val noLvlDrawable: Drawable = ...
    init {
        ...
        scene2d.table {
            ...
            for (i in 1..10) {
                image(if (i <= mod.level) lvlDrawable else noLvlDrawable) {it.size(4f)} // this line doesn't work now
            }
        }
    }
}
```
It shows this messages:
![image](https://user-images.githubusercontent.com/25386431/89501843-b351bc80-d7cc-11ea-933c-fe9d63833159.png)
![image](https://user-images.githubusercontent.com/25386431/89501899-cc5a6d80-d7cc-11ea-99d0-bf279cb72ec7.png)

But when I define these variables in table scope - it works fine. I guess there may be some conflict with contracts, but can't understand what is it and how to fix it.

**Comment by czyzby** [comment]
I don't think it's related strictly to contracts, it's more of a DSL issue. I don't see why it wouldn't work though, have you tried rebuilding the file? Cleaning and rebuilding the project might help. I copied the code and it compiles just fine:

![image](https://user-images.githubusercontent.com/11707612/89526942-d1c5b100-d7e8-11ea-9051-c419d3ed4aed.png)

As for fixing that particular error, you can use explicit `this@ModIcon` receiver. For example: `this@ModIcon.mod`. However, this should not be necessary.

**Comment by Divelix** [comment, marked as answer]
Oh, I just checked again, actually my code was:
```kotlin
@Scene2dDsl
class ModIcon(val mod: Mod, ...): Group(), KGroup {...}
// I implemented KGroup to use it as my custom widget
modIcon(mod) {...}
```
and I did not pay attention to that when wrote the code in the question. It seems like I broke smth inside `KGroup` by that code, because `ModIcon` works fine now without inheritance from `KGroup`.

**Comment by czyzby** [comment]
Glad you worked it out.
